### PR TITLE
fix: change shuffle pin state after shuffle toggle is disabled

### PR DESCRIPTION
**Pull request by Karol-Stelmaczonek**
Ticket: https://oat-sa.atlassian.net/browse/AUT-3008

## What's Changed

- Disabling shuffle choices checkbox in authoring item also removes changes on shuffle pins

## How to test
- Open item in Authoring mode
- Tick “Shuffle choices“ option 
- Pin all choices and save
- Tick off “Shuffle choices“ option  and save
- Tick “Shuffle choices“ option again
- Observe Shuffling interface

image

### CODE REVIEW

**Review by albertosgz — COMMENTED** [review]
To me looks good and clear. Can you check out my comment? just in case, avoid a possible error.

File: views/js/qtiCreator/widgets/choices/helpers/formElement.js [hunk 1]
```diff
@@ -28,6 +28,14 @@ define(['jquery'], function ($) {
                     $shuffleToggle.show();
                 } else {
                     $shuffleToggle.hide();
+                    let icon = $shuffleToggle.children();
+                    if (icon.length === 0) {
```
[the remark this answers is not in the document]
Comment (albertosgz) [reply views/js/qtiCreator/widgets/choices/helpers/formElement.js:32]: No problem. Happy to help ;)

**Review by bziondik — CHANGES_REQUESTED** [review]
Idea is right, but could your improve code? you have duplication with lines 49-54
move this part in separate function

File: views/js/qtiCreator/widgets/choices/helpers/formElement.js [hunk 2]
```diff
@@ -23,6 +23,16 @@ define(['jquery'], function ($) {
                 interaction = choice.getInteraction(),
                 $shuffleToggle = $container.find('[data-role="shuffle-pin"]');
 
+            const pinIcon = function (icon) {
```
Comment (bziondik) [inline views/js/qtiCreator/widgets/choices/helpers/formElement.js:26]: better use name `$icon` it is our code style -use $ when it is jQuery element

**Review by bziondik — APPROVED** [review]
- [ ]  New code is covered by tests (if applicable)
- [x]  Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x]  New code is respecting code style rules
- [x]  New code is respecting best practices
- [x]  New code is not subject to concurrency issues (if applicable)
- [x]  Feature is working correctly on my local machine (if applicable)
- [x]  Acceptance criteria are respected
- [x]  Pull request title and description are meaningful

File: views/js/qtiCreator/widgets/choices/helpers/formElement.js (hunk 2 above)
Comment (Karol-Stelmaczonek) [reply views/js/qtiCreator/widgets/choices/helpers/formElement.js]: done

**Pull request merged**